### PR TITLE
Update button background color to ensure storybook examples meet contrast

### PR DESCRIPTION
**Pull request by khiga8**
Closes # https://github.com/github/accessibility-audits/issues/5561

#### New

I updated the background colors for button used in a story. The intent of the story seems to be to demonstrate that users can customize the component via `sx` prop.

I followed what @lindseywild did in https://github.com/primer/react/pull/4528.

### Rollout strategy

- [ ] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [x] None; if selected, include a brief description as to why

This only impacts storybook stories.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/Button/Button.dev.stories.tsx [hunk 1]
```diff
@@ -53,20 +53,17 @@ export const TestSxProp = () => {
           },
           [`@media screen and (max-width: 768px)`]: {
             color: 'maroon',
-          },
-          '@media screen and (min-width: 768px)': {
```
Comment (khiga8) [inline packages/react/src/Button/Button.dev.stories.tsx:57]: I understand the intent of these examples is to show that consumers can customize their buttons via `sx` prop. However, I also don't think this (customizing focus) is good practice we want to endorse. It is easier for me to just remove the example.

File: packages/react/src/Button/Button.dev.stories.tsx [hunk 2]
```diff
@@ -53,20 +53,17 @@ export const TestSxProp = () => {
...
-            ':focus': {
-              color: 'green',
-            },
+            backgroundColor: '#F6F8FA',
```
Comment (khiga8) [inline packages/react/src/Button/Button.dev.stories.tsx:56]: I copied approach in https://github.com/primer/react/pull/4528.

File: contributor-docs/testing.md [hunk 3]
```diff
@@ -132,6 +132,9 @@ available, check out the table below.
 > However, screenshots will not match and new ones will need to be generated on
 > your first test run.
 
+> [!TIP]
```
Comment (khiga8) [inline contributor-docs/testing.md:135]: Documenting this awesome pro-tip that Josh informed me about!

Comment (lindseywild) [reply contributor-docs/testing.md:135]: Thank you for adding this!

**Review by lindseywild — APPROVED** [review]
Looks good from accessibility, but it definitely still needs a Primer person's review!

**Review by joshblack — APPROVED** [review]

**Review by langermank — APPROVED** [review]

**Pull request merged**